crates/ide/src/syntax_highlighting/inject.rs

```diff
@@ -90,6 +90,13 @@ const RUSTDOC_FENCE_TOKENS: &[&'static str] = &[
     "edition2021",
 ];
 
+fn is_rustdoc_fence_token(token: &str) -> bool {
+    if RUSTDOC_FENCE_TOKENS.contains(&token) {
```

Comment:
Let’s just test if token *starts* with compile_fail, etc, instead of doing a strict match? That way, we’ll correctly handle incomplete code.

Reply:
Oh damn I completely missed that, I'll fix that up

Reply:
Might make sense to check if we have “be resilient to incorrect input” in the cross cutting concerns in architecture.md

Reply:
Actually dont you mean whether one of the tokens inside `RUSTDOC_FENCE_TOKENS` starts with the given token? as in the other way around? Or am I not seeing what you mean with incomplete code here

Reply:
I might be misreading logic here, but, basically, we should identify compile_error,ENope as Rust. The current code I believe won’t do this.

Reply:
Oh yes you are right, we are requiring all of them to fit currently which does seem like a bad idea